internal/service/iam/policy_model.go

```diff
@@ -227,3 +231,57 @@ func policyDecodeConfigStringList(lI []interface{}) interface{} {
 	sort.Sort(sort.Reverse(sort.StringSlice(ret)))
 	return ret
 }
+
+// PolicyHasValidAWSPrincipals validates that the Principals in an IAM Policy are valid
+// Assumes that non-"AWS" Principals are valid
+// The value can be a single string or a slice of strings
+// Valid strings are either an ARN or an AWS account ID
+func PolicyHasValidAWSPrincipals(policy string) (bool, error) { // nosemgrep:ci.aws-in-func-name
```

Comment:
Nit - Maybe renaming to something like `PolicyHasValidOrZeroAWSPrincipals` could make it more obvious policies with non-"AWS" principals return `true`.